src/panels/lovelace/components/hui-action-editor.ts

```diff
@@ -0,0 +1,129 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import "@polymer/paper-input/paper-textarea";
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-listbox/paper-listbox";
+
+import "../../../components/ha-service-picker";
+
+import { HomeAssistant } from "../../../types";
+import { fireEvent, HASSDomEvent } from "../../../common/dom/fire_event";
+import { EditorTarget } from "../editor/types";
+import {
+  ActionConfig,
+  NavigateActionConfig,
+  CallServiceActionConfig,
+} from "../../../data/lovelace";
+
+declare global {
+  // for fire event
+  interface HASSDomEvents {
+    "action-changed": undefined;
+  }
+  // for add event listener
+  interface HTMLElementEventMap {
+    "action-changed": HASSDomEvent<undefined>;
+  }
+}
+
+export class HuiActionEditor extends LitElement {
+  public config?: ActionConfig;
+  public label?: string;
+  public actions?: string[];
+  protected hass?: HomeAssistant;
+
+  static get properties(): PropertyDeclarations {
+    return { hass: {}, config: {} };
+  }
+
+  get _action(): string {
+    return this.config!.action || "";
+  }
+
+  get _navigation_path(): string {
+    const config = this.config! as NavigateActionConfig;
+    return config.navigation_path || "";
+  }
+
+  get _service(): string {
+    const config = this.config! as CallServiceActionConfig;
+    return config.service || "";
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.actions) {
+      return html``;
+    }
+    return html`
+      <paper-dropdown-menu
+        .label="${this.label}"
+        .configValue="${"action"}"
+        @value-changed="${this._valueChanged}"
+      >
+        <paper-listbox
+          slot="dropdown-content"
+          .selected="${this.actions.indexOf(this._action)}"
+        >
+          ${
+            this.actions.map((action) => {
+              return html`
+                <paper-item>${action}</paper-item>
```

Comment:
When I change the item, I expect that the keys for the other items to be dropped. We don't want to set a navigation_path when we change to service call?